lib/tapioca/compilers/dsl/active_record_columns.rb

```diff
@@ -0,0 +1,347 @@
+            methods_to_add,
+            return_type: getter_type
+          )
+          add_method(
+            klass,
+            "saved_change_to_#{column_name}",
+            methods_to_add,
+            return_type: "[#{getter_type}, #{getter_type}]"
+          )
+          add_method(
+            klass,
+            "saved_change_to_#{column_name}?",
+            methods_to_add,
+            return_type: "T::Boolean"
+          )
+          add_method(
+            klass,
+            "will_save_change_to_#{column_name}?",
+            methods_to_add,
+            return_type: "T::Boolean"
+          )
+
+          # Added by ActiveModel::Dirty
+          #
+          add_method(
+            klass,
+            "#{column_name}_change",
+            methods_to_add,
+            return_type: "[#{getter_type}, #{getter_type}]"
+          )
+          add_method(
+            klass,
+            "#{column_name}_changed?",
+            methods_to_add,
+            return_type: "T::Boolean"
+          )
+          add_method(
+            klass,
+            "#{column_name}_will_change!",
+            methods_to_add
+          )
+          add_method(
+            klass,
+            "#{column_name}_was",
+            methods_to_add,
+            return_type: getter_type
+          )
+          add_method(
+            klass,
+            "#{column_name}_previous_change",
+            methods_to_add,
+            return_type: "[#{getter_type}, #{getter_type}]"
+          )
+          add_method(
+            klass,
+            "#{column_name}_previously_changed?",
+            methods_to_add,
+            return_type: "T::Boolean"
+          )
+          add_method(
+            klass,
+            "#{column_name}_previously_was",
+            methods_to_add,
+            return_type: getter_type
+          )
+          add_method(
+            klass,
+            "restore_#{column_name}!",
+            methods_to_add
+          )
+
+          # Added by ActiveRecord::AttributeMethods::BeforeTypeCast
+          #
+          add_method(
+            klass,
+            "#{column_name}_before_type_cast",
+            methods_to_add,
+            return_type: "T.untyped"
+          )
+          add_method(
+            klass,
+            "#{column_name}_came_from_user?",
+            methods_to_add,
+            return_type: "T::Boolean"
+          )
+        end
+
+        sig do
+          params(
+            constant: T.class_of(ActiveRecord::Base),
+            column_name: String
+          ).returns([String, String])
+        end
+        def type_for(constant, column_name)
+          column_type = constant.attribute_types[column_name]
+
+          if Object.const_defined?(:StrongTypeGeneration) &&
+              !(constant.singleton_class < Object.const_get(:StrongTypeGeneration))
+            return ["T.untyped", "T.untyped"]
+          end
+
+          getter_type =
+            case column_type
```

Comment:
Can we test each of these cases please? It can be a single test with columns of each of these types and we can assert the inclusion of the `field=` method signatures, which would tell us if we are mapping the types correctly, or not.